src/python/pants/invalidation/build_invalidator.py

```diff
@@ -93,8 +94,30 @@ def key_for_target(self, target, transitive=False, fingerprint_strategy=None):
 class BuildInvalidator(object):
   """Invalidates build targets based on the SHA1 hash of source files and other inputs."""
 
-  def __init__(self, root):
-    self._root = os.path.join(root, GLOBAL_CACHE_KEY_GEN_VERSION)
+  class Factory(Subsystem):
+    options_scope = 'build-invalidator'
+
```

Comment:
Maybe the `--force` flag itself should be defined here instead of in global scope?  We can play with the scope name to make the flag prefixes make sense, e.g.,

`--invalidate-force`, `--invalidate-compile-java-force`